engine/consensus/dkg/reactor_engine.go

```diff
@@ -161,33 +229,56 @@ func (e *ReactorEngine) EpochSetupPhaseStarted(currentEpochCounter uint64, first
 	e.registerPhaseTransition(curDKGInfo.phase3FinalView, dkgmodule.Phase3, e.end(nextEpochCounter))
 }
 
-// EpochCommittedPhaseStarted handles the EpochCommittedPhaseStarted protocol event. It
-// compares the key vector locally produced by Consensus nodes against the FlowDKG smart contract
-// key vectors. If the keys don't match a log statement will be invoked. In the happy case the locally
-// produced key should match, if the keys do not match the node will have a invalid random beacon key.
-func (e *ReactorEngine) EpochCommittedPhaseStarted(currentEpochCounter uint64, first *flow.Header) {
+// handleEpochCommittedPhaseStarted checks that the local DKG completed and
+// that our locally computed key share is consistent with the canonical key vector.
+//
+// If our key is inconsistent, we log a warning message.
+// CAUTION: we will still sign using the inconsistent key, to be fixed with Crypto
+// v2 by falling back to staking signatures in this case.
+//
+func (e *ReactorEngine) handleEpochCommittedPhaseStarted(currentEpochCounter uint64) {
+
+	// the epoch counter the DKG just completed is for
+	dkgEpochCounter := currentEpochCounter + 1
+
+	// TODO - first check whether we've already stored the dkg end state, since
+	// we need to handle multiple calls to this method
+
+	// TODO execute in transaction scope of protocol event emitter, need to use
+	// lower-level storage access here (since State doesn't allow tx injection)
 	nextDKG, err := e.State.Final().Epochs().Next().DKG()
 	if err != nil {
-		e.log.Err(err).Msg("checking DKG key consistency: could not retrieve next DKG info")
+		// CAUTION: this should never happen, indicates a storage failure or corruption
+		e.log.Fatal().Err(err).Msg("checking DKG key consistency: could not retrieve next DKG info")
 		return
 	}
 
-	dkgPrivInfo, err := e.keyStorage.RetrieveMyBeaconPrivateKey(currentEpochCounter + 1)
-	if err != nil {
-		e.log.Err(err).Msg("checking DKG key consistency: could not retrieve DKG private info for next epoch")
+	myBeaconPrivKey, err := e.dkgState.RetrieveMyBeaconPrivateKey(currentEpochCounter + 1)
+	if errors.Is(err, storage.ErrNotFound) {
+		e.log.Warn().Msg("checking DKG key consistency: no key found")
+		err := e.dkgState.SetDKGEndState(dkgEpochCounter, flow.DKGEndStateNoKey)
```

Comment:
Definitely, was planning to address that in the next PR, since that storage read would also need to make use of the badger transaction context we're passing in.